File: UI/wwwroot/Facturacion/FrontModel/Cat_Producto.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { Cat_Marca } from './Cat_Marca.js'
import { Cat_Categorias } from './Cat_Categorias.js'
import { Detalle_Compra } from './Detalle_Compra.js'
import { Detalle_Factura } from './Detalle_Factura.js'
class Cat_Producto extends EntityClass {
    constructor(props) {
        super(props, 'EntityFacturacion');
        Object.assign(this, props);
    }
   /**@type {Number}*/ Id_Producto;
   /**@type {String}*/ Descripcion;
   /**@type {Cat_Marca} ManyToOne*/ Cat_Marca;
   /**@type {Cat_Categorias} ManyToOne*/ Cat_Categorias;
   /**@type {Array<Detalle_Compra>} OneToMany*/ Detalle_Compra;
   /**@type {Array<Detalle_Factura>} OneToMany*/ Detalle_Factura;
}
export { Cat_Producto }

File: BusinessLogic/Facturacion/Mapping/Cat_Producto.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using APPCORE;
namespace DataBaseModel
{
    public class Cat_Producto : EntityClass
    {
        [PrimaryKey(Identity = true)]
        public int? Id_Producto { get; set; }
        public string? Descripcion { get; set; }
        public int? Id_Categoria { get; set; }
        public int? Id_Marca { get; set; }
        [ManyToOne(TableName = "Cat_Marca", KeyColumn = "Id_Marca", ForeignKeyColumn = "Id_Marca")]
        public Cat_Marca? Cat_Marca { get; set; }
        [ManyToOne(TableName = "Cat_Categorias", KeyColumn = "Id_Categoria", ForeignKeyColumn = "Id_Categoria")]
        public Cat_Categorias? Cat_Categorias { get; set; }
        //[OneToMany(TableName = "Detalle_Compra", KeyColumn = "Id_Producto", ForeignKeyColumn = "Id_Producto")]
        public List<Detalle_Compra>? Detalle_Compra { get; set; }
        //[OneToMany(TableName = "Detalle_Factura", KeyColumn = "Id_Producto", ForeignKeyColumn = "Id_Producto")]
        public List<Detalle_Factura>? Detalle_Factura { get; set; }

        public static void SetProductData(Cat_Producto? productParam)
        {
            Cat_Producto? producto = productParam?.Find<Cat_Producto>();            
            if (producto != null)
            {
               productParam?.Save();
            }  
            
        }
    }
}
